cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -101,4 +101,16 @@ type NodeConfiguration struct {
 	NodeName                 string
 	TLSBootstrapToken        string
 	Token                    string
+
+	// TLSDiscoveryRootCAPubKeys specifies a set of public key pins. The root CA
+	// discovered during TLS bootstrapping must match one of these values.
+	// Specifying an empty set disables root CA pinning. Each hash is a hex encoded
```

Comment:
I tried to implement the `pubkeypin` so it's easy to swap in support for multiple encodings and lengths.

I think if we do that we should also switch the default token encodings as well.